### PR TITLE
7.0 account voucher remit to

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This PR substitutes https://github.com/OCA/account-payment/pull/21.

In countries where payments are made by checks the account vouchers are used, and it often happens that the addresses that the checks should be sent to is different to the address of the supplier.

The new modules included in this PR add a new address type on the partner 'Remit to'.
During the creation of a Voucher, once the partner is entered the partner's remit-to is fetched and added to the voucher.

See a blog post that explains it:
http://goo.gl/KRbdJW

Regards,
Jordi Ballester

### CODE REVIEW

File: account_check_writing_remit_to/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline account_check_writing_remit_to/__init__.py:1]: Thanks for your contribution to the OCA. 
Please simplify your headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: account_check_writing_remit_to/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Check Writing using Remit-to address',
+    'version': '1.0',
+    'author': 'Eficent, Odoo Community Association (OCA)',
+    'category': 'Generic Modules/Accounting',
+    'description': """
```
Comment (elicoidal) [inline account_check_writing_remit_to/__openerp__.py]: Please move the description to the README file

File: account_check_writing_remit_to/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Check Writing using Remit-to address',
+    'version': '1.0',
```
Comment (elicoidal) [inline account_check_writing_remit_to/__openerp__.py]: Please follow the guidelines for the version (8.0.1.0.0) from:
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

File: account_check_writing_remit_to/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    'author': 'Eficent, Odoo Community Association (OCA)',
+    'category': 'Generic Modules/Accounting',
+    'description': """
+Check Writing using Remit-to address
+====================================
+Module to show the remit-to address during the Check Printing.
```
Comment (elicoidal) [inline account_check_writing_remit_to/__openerp__.py]: Remove unnecessary capitals

File: account_check_writing_remit_to/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
+    """,
+    'website': 'http://www.eficent.com',
+    'depends': ['account_check_writing', 'account_voucher_remit_to'],
+    'data': [
+        'account_check_writing_report.xml',
+    ],
+    'demo': [],
```
Comment (elicoidal) [inline account_check_writing_remit_to/__openerp__.py]: remove empty keys

File: account_check_writing_remit_to/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    'test': [],
+    'installable': True,
+    'active': False,
```
Comment (elicoidal) [inline account_check_writing_remit_to/__openerp__.py]: Not sure about this key

File: account_check_writing_remit_to/report/__init__.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import check_print
```
Comment (elicoidal) [inline account_check_writing_remit_to/report/__init__.py]: Add line

File: account_check_writing_remit_to/report/check_print_bottom.rml [hunk 8]
```diff
@@ -0,0 +1,317 @@
+<?xml version="1.0"?>
+<document filename="test.pdf">
+  <template pageSize="(595, 842)" title="Test" author="Martin Simon" allowSplitting="20">
```
Comment (elicoidal) [inline account_check_writing_remit_to/report/check_print_bottom.rml]: Modify information accordingly (filename, title, author)

File: account_check_writing_remit_to/report/check_print_middle.rml [hunk 9]
```diff
@@ -0,0 +1,355 @@
+<?xml version="1.0"?>
+<document filename="test.pdf">
```
Comment (elicoidal) [inline account_check_writing_remit_to/report/check_print_middle.rml:2]: Modify header information accordingly

File: account_check_writing_remit_to/report/check_print_top.rml [hunk 10]
```diff
@@ -0,0 +1,335 @@
+<?xml version="1.0"?>
+<document filename="test.pdf">
+  <template pageSize="(595, 842)" title="Test" author="Martin Simon" allowSplitting="20">
```
Comment (elicoidal) [inline account_check_writing_remit_to/report/check_print_top.rml:3]: Modify header information accordingly

File: account_voucher_payment_method/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline account_voucher_payment_method/__openerp__.py]: Better split your PR one per module, it helps the review

File: account_voucher_payment_method/__openerp__.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+##############################################################################
```
Comment (elicoidal) [inline account_voucher_payment_method/__openerp__.py]: Please simplify headers

File: account_voucher_payment_method/__openerp__.py [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+{
+    "name": "Account Voucher Payment Method",
+    "version": "1.0",
```
Comment (elicoidal) [inline account_voucher_payment_method/__openerp__.py]: Version, descriptions, empty keys

File: account_voucher_payment_method/i18n/account_voucher_remit_to.po [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+msgstr ""
+
+#. module: account_voucher_remit_to
+#: model:ir.model,name:account_voucher_remit_to.model_account_voucher
+msgid "Accounting Voucher"
+msgstr ""
+
+#. module: account_voucher_remit_to
+#: view:account.voucher:0
+msgid "onchange_partner_id(partner_id, remit_to, journal_id, amount, currency_id, type, date, context)"
```
Comment (elicoidal) [inline account_voucher_payment_method/i18n/account_voucher_remit_to.po]: This should not be in the po file

File: account_voucher_remit_to/__openerp__.py [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+{
+    "name": "Account Voucher Remit-to address",
+    "version": "1.0",
```
Comment (elicoidal) [inline account_voucher_remit_to/__openerp__.py]: version/description/README

File: account_voucher_remit_to/model/res_partner.py [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+class res_partner(osv.osv):
+    _inherit = "res.partner"
+
+    _columns = {
+        'type': fields.selection(
+            [('default', 'Default'), ('invoice', 'Invoice'),
+             ('delivery', 'Shipping'), ('contact', 'Contact'),
+             ('other', 'Other'), ('remit_to', 'Remit-to')],
+            'Address Type',
+            help="Used to select automatically the right address "
```
Comment (elicoidal) [inline account_voucher_remit_to/model/res_partner.py]: s/right/suitable

File: account_voucher_remit_to/models/res_partner.py [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+class ResPartner(orm.Model):
+    _inherit = "res.partner"
+
+    _columns = {
+        'type': fields.selection(
+            [('default', 'Default'), ('invoice', 'Invoice'),
+             ('delivery', 'Shipping'), ('contact', 'Contact'),
+             ('other', 'Other'), ('remit_to', 'Remit-to')],
+            'Address Type',
+            help="Used to select automatically the right address "
```
Comment (elicoidal) [inline account_voucher_remit_to/models/res_partner.py:16]: s/right/suitable

File: account_voucher_remit_to/view/account_voucher_view.xml [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+
+        <!-- This general view is used in 
+             Invoicing - Journal Entries - Journal Vouchers -->
+        <record model="ir.ui.view" id="view_voucher_form">
+            <field name="name">account.voucher.form</field>
+            <field name="model">account.voucher</field>
+            <field name="inherit_id" ref="account_voucher.view_voucher_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='partner_id']" position="attributes">
+                    <attribute name="on_change">onchange_partner_id(partner_id, remit_to, journal_id, amount, currency_id, type, date, context)</attribute>
```
Comment (elicoidal) [inline account_voucher_remit_to/view/account_voucher_view.xml:23]: To be checked?

File: account_voucher_remit_to/models/res_partner.py (hunk 17 above)
Comment (JordiBForgeFlow) [inline account_voucher_remit_to/models/res_partner.py:16]: What does "s/right/suitable" mean?

File: account_voucher_remit_to/view/account_voucher_view.xml (hunk 18 above)
Comment (JordiBForgeFlow) [inline account_voucher_remit_to/view/account_voucher_view.xml:23]: the on_change is needed to redetermine the remit_to when the partner changes.

**Comment by JordiBForgeFlow** [comment]
Thanks @elicoidal commited changes as per your comments

**Comment by pedrobaeza** [comment]
As everything is red and this version is very old, I close this PR.

**Pull request closed without merging**